src/services/ApiService.ts

```diff
@@ -12,24 +12,30 @@ const getPort = () => {
 }
 
 /**
- * API client to make requests to the endpoints and passing the API_KEY for authentication.
+ * API client to make requests to the endpoints and passing the JWT for authentication.
+ * Start as null and is initialized by App.vue
  */
-export const apiClient = new CatClient({
-	baseUrl: window.location.hostname,
-	port: getPort(),
-	secure: window.location.protocol === 'https:',
-	timeout: 15000,
-	instant: false,
-	ws: {
-		retries: 3,
-		delay: 3000,
-		onFailed: () => {
-			console.error('Failed to connect WebSocket after 3 retries.')
-		},
-	},
-})
+let apiClient: CatClient | null = null
+export { apiClient }
+
 
-export const updateCredential = (cred: string | undefined) => (apiClient.credential = cred)
+export const instantiateApiClient = (credential: string | undefined) => {
+	apiClient = new CatClient({
+		baseUrl: window.location.hostname,
+		port: getPort(),
+		secure: window.location.protocol === 'https:',
+		credential: credential,
+		timeout: 15000,
+		instant: true,
+		ws: {
+			retries: 1000, // big number here because the count is cumulative (`retries` is not reset on connection)
```

Comment:
i'm not sure i understood what you said, what do you mean by `reset on connection`? after the ws is connected it won't retry anymore until you close and reopen the ws with something like `apiClient.reset().init()`

Reply:
I mean after the automatic reconnection, retries is not reset to 0

Reply:
Not able to fix this at the moment :(

Reply:
I fixed it on the api client package, now it resets the retries on connection open

Reply:
Awesome 🪽